packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -65,17 +65,32 @@ import {
  */
 export class BlobHandle
 	extends FluidHandleBase<ArrayBufferLike>
-	implements IFluidHandleInternalPayloadPending<ArrayBufferLike>
+	implements
+		IFluidHandlePayloadPending<ArrayBufferLike>,
+		IFluidHandleInternalPayloadPending<ArrayBufferLike>
 {
 	private attached: boolean = false;
 
 	public get isAttached(): boolean {
 		return this.routeContext.isAttached && this.attached;
 	}
 
+	private _events:
+		| (Listenable<IFluidHandlePayloadPendingEvents> &
+				IEmitter<IFluidHandlePayloadPendingEvents>)
+		| undefined;
+	public get events(): Listenable<IFluidHandlePayloadPendingEvents> {
+		return (this._events ??= createEmitter<IFluidHandlePayloadPendingEvents>());
+	}
+
+	private _state: PayloadState = "local";
```

Comment:
what will this do on a remote client right now?

Reply:
Nothing since remote clients don't instantiate BlobHandles (they will only have RemoteFluidObjectHandles).  When we add support in RemoteFluidObjectHandles they'll have their own corresponding state/events for "pending" -> "shared".

Reply:
so isFluidHandlePayloadPending will always return false for a remote blob handle?

Reply:
Not in the future but yes for this PR (since we haven't implemented observability for remote handles yet).  It will always return true when the payload status is inspectable/observable.

Once observability is moved to IFluidHandle then the typeguard isn't needed anymore and can just be removed.